util/make_new_dif.py

```diff
@@ -115,8 +115,9 @@ def _load_irqs(self):
         assert (self.hjson_data and
                 "ERROR: must load IP HJSON before loarding IRQs")
         irqs = []
-        for irq in self.hjson_data["interrupt_list"]:
-            irqs.append(Irq(irq))
+        if "interrupt_list" in self.hjson_data:
```

Comment:
Nit: (L131) 

Would you please be able to add a more extended explanation on what "batch" and "single" means?

`"mode to generate DIF code; use 'single' if no DIF code exists."` - maybe a use case for "single" and "batch" could be useful to add to the description.

Reply:
Apologies for the above, I know it is outside this particular commit, but I assume the above changes have been already merged in. So this is a way, a post merge comment :)

Reply:
Nit:

with `util/make_new_dif.py --ip sram_ctrl --peripheral "SRAM Controller" -m single`

It generates `sw/device/lib/dif/autogen/dif_sram_ctrl_autogen.h` but in `sw/device/lib/dif/dif_sram_ctrl.h`, it includes `#include "sw/device/lib/dif/autogen/dif_sram_ctrl_autogen.inc" (which doesn't exist, but I assume `dif_sram_ctrl_autogen.h` is supposed to be `.inc`)?

Reply:
Thanks for the suggestions @silvestrst ! I will roll these into a separate PR, if thats ok? since they are orthogonal to this PR if you mention above. My apologies for the lack of clarity in the usage commentary.

Also the issue you mention with the .inc/.h, this was because the `util/make_new_dif/dif_template.h.tpl` was not updated to match the change from .inc to .h in #8213. I have updated #8213 and rebased this PR on top (since it is dependent). If you do the same and re-run the `util/make_new_dif.py` script as you describe above this should be fixed. Sorry for the confusion!

Reply:
> Thanks for the suggestions @silvestrst ! I will roll these into a separate PR, if thats ok? since they are orthogonal to this PR if you mention above.

Whatever you think is the best, thank you for further automating DIF generation!

> My apologies for the lack of clarity in the usage commentary.

No worries, that's what reviews are for! :)